File: EasyTable/Demos/C++ Builder 6/Batch/uBatch.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "uBatch.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "EasyTable"
#pragma resource "*.dfm"
TForm1 *Form1;
//---------------------------------------------------------------------------
__fastcall TForm1::TForm1(TComponent* Owner)
        : TForm(Owner)
{
}
//---------------------------------------------------------------------------


void __fastcall TForm1::FormCreate(TObject *Sender)
{
 SourceTable->Open();
 DestTable->FieldDefs->Assign(SourceTable->FieldDefs);
 DestTable->IndexDefs->Assign(SourceTable->IndexDefs);
 DestTable->CreateTable();
 DestTable->Open();
}
//---------------------------------------------------------------------------
void __fastcall TForm1::Button1Click(TObject *Sender)
{
 AnsiString Log;

 switch (rgMode->ItemIndex)
  {
   case 0 : if (! DestTable->AddRecords(SourceTable, arAppend, Log))
    { MessageDlg("Error adding records: "+Log,mtError,TMsgDlgButtons()<<mbOK,0);}
   case 1 : if (! DestTable->AddRecords(SourceTable, arUpdate, Log))
    { MessageDlg("Error adding records: "+Log,mtError,TMsgDlgButtons()<<mbOK,0);}
   case 2 : if (! DestTable->AddRecords(SourceTable, arAppendUpdate, Log))
    { MessageDlg("Error adding records: "+Log,mtError,TMsgDlgButtons()<<mbOK,0);}
   case 3 : if (! DestTable->AddRecords(SourceTable, arReplace, Log))
    { MessageDlg("Error adding records: "+Log,mtError,TMsgDlgButtons()<<mbOK,0);}
  }
}
//---------------------------------------------------------------------------

File: EasyTable/Demos/Delphi/Batch/uBatch.pas
unit uBatch;

interface

uses
  Windows, Messages, SysUtils, Classes, Graphics, Controls, Forms, Dialogs,
  Db, EasyTable, StdCtrls, ExtCtrls, DBCtrls, Grids, DBGrids;

type
  TForm1 = class(TForm)
    Label1: TLabel;
    DataSource1: TDataSource;
    DataSource2: TDataSource;
    DBGrid1: TDBGrid;
    DBGrid2: TDBGrid;
    DBNavigator1: TDBNavigator;
    DBNavigator2: TDBNavigator;
    Button1: TButton;
    rgMode: TRadioGroup;
    SourceTable: TEasyTable;
    DestTable: TEasyTable;
    procedure Button1Click(Sender: TObject);
    procedure FormCreate(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  Form1: TForm1;

implementation

{$R *.DFM}

procedure TForm1.Button1Click(Sender: TObject);
var Log : string;
begin
 case rgMode.ItemIndex of
  0: if (not DestTable.AddRecords(SourceTable, arAppend, Log)) then
     MessageDlg('Error adding records: '+Log,mtError,[mbOk],0);
  1: if (not DestTable.AddRecords(SourceTable, arUpdate, Log)) then
     MessageDlg('Error adding records: '+Log,mtError,[mbOk],0);
  2: if (not DestTable.AddRecords(SourceTable, arAppendUpdate, Log)) then
     MessageDlg('Error adding records: '+Log,mtError,[mbOk],0);
  3: if (not DestTable.AddRecords(SourceTable, arReplace, Log)) then
     MessageDlg('Error adding records: '+Log,mtError,[mbOk],0);
 end;
end;

procedure TForm1.FormCreate(Sender: TObject);
begin
 SourceTable.Open;
 DestTable.FieldDefs.Assign(SourceTable.FieldDefs);
 DestTable.IndexDefs.Assign(SourceTable.IndexDefs);
 DestTable.CreateTable;
 DestTable.Open
end;

end.
